edi_oca/models/edi_backend.py

```
@@ -634,10 +540,11 @@ def _exchange_receive_check(self, exchange_record):
         ]
 
     def _exchange_receive(self, exchange_record):
-        component = self._get_component(exchange_record, "receive")
-        if component:
-            return component.receive()
-        raise NotImplementedError()
+        exchange_function = self._get_exec_handler(exchange_record, "receive")
+        if exchange_function:
+            ctx = self._get_record_env_ctx(exchange_record, "receive")
+            return exchange_function(exchange_record.with_context(**ctx))
+        raise EDINotImplementedError("No handler for `receive`")
```

to simplify: `_get_exec_handler` should raise if not found